src/cgw_ucentral_messages_queue_manager.rs

```diff
@@ -0,0 +1,360 @@
+
+        device_msg_queue.get_state()
+    }
+
+    pub async fn set_device_queue_state(
+        &self,
+        device_mac: &String,
+        state: CGWUCentralMessagesQueueState,
+    ) {
+        let container_lock = self.queue.read().await;
+        let mut device_msg_queue = container_lock.get(device_mac).unwrap().write().await;
+
+        device_msg_queue.set_state(state);
+    }
+
+    pub async fn insert_disconnected_device_timeout(&self, device_mac: String, seconds: u64) {
+        let mut disconnected_lock = self.disconnected_devices.write().await;
+        disconnected_lock.insert(device_mac, Duration::new(seconds, 0));
+    }
+
+    pub async fn remove_disconnected_device_timeout(&self, device_mac: &String) {
+        let mut disconnected_lock = self.disconnected_devices.write().await;
+        disconnected_lock.remove(device_mac);
+    }
+
+    pub async fn iterate_over_disconnected_devices(&self) {
+        let container_read_lock = self.disconnected_devices.read().await;
+        let mut update_internal_list: Vec<String> = Vec::<String>::new();
+        // 1. Iterate over disconnected devices and
+        // remove message queue if it is already empty
+        // or decrement timeout value
+        for (device_mac, _) in container_read_lock.iter() {
+            if self.get_device_messages_queue_len(device_mac).await > 0 {
+                self.decrement_disconnected_device_timeout(device_mac, 10)
+                    .await;
+            } else {
+                self.delete_device_messages_queue(device_mac).await;
+                update_internal_list.push(device_mac.clone());
+            }
+        }
+
+        // 2. Update disconnected device list
+        for device_mac in update_internal_list {
+            self.remove_disconnected_device_timeout(&device_mac).await;
+        }
+
+        // 3. Iterate over disconnected devices and get timed out devices
+        let devices_to_remove: Vec<String> = container_read_lock
+            .iter()
+            .filter_map(|(device_mac, duration)| {
+                if duration.as_secs() <= 0 {
+                    Some(device_mac.clone())
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        // 4. Iterate over timedout devices and remove queues
+        let mut update_internal_list: Vec<String> = Vec::<String>::new();
+        for device_mac in devices_to_remove.iter() {
+            debug!("Remove device {} message queue due to timeout!", device_mac);
+            self.delete_device_messages_queue(device_mac).await;
+            update_internal_list.push(device_mac.clone());
+        }
+
+        // 5. Update disconnected device list
+        for device_mac in update_internal_list {
+            self.remove_disconnected_device_timeout(&device_mac).await;
+        }
+    }
+
+    pub async fn decrement_disconnected_device_timeout(&self, device_mac: &String, seconds: u64) {
```

Comment:
I'm thinking maybe we should consider using Duration? It would be safer, also there are underflow checks present which is useful to track whether timeout's elapsed;